Sources/SwiftSyntax/SyntaxParser.swift

```diff
@@ -107,9 +117,11 @@ public struct SyntaxParser {
   private static func parseRaw(
     _ source: String,
     parseLookup: IncrementalParseLookup?
-  ) -> RawSyntax {
+  ) throws -> RawSyntax {
```

Comment:
I have a minor preference to do the check and throw the error in `parse(source:, parseLookup:)` function so that we keep all the throwing of errors within one function.

Reply:
Done